modules/surface_matching/src/icp.cpp

```diff
@@ -540,12 +540,22 @@ int ICP::registerModelToScene(const Mat& srcPC, const Mat& dstPC, double& residu
 // source point clouds are assumed to contain their normals
 int ICP::registerModelToScene(const Mat& srcPC, const Mat& dstPC, std::vector<Pose3DPtr>& poses)
 {
-  for (size_t i=0; i<poses.size(); i++)
+  #if defined _OPENMP
```

Comment:
In OpenCV there is own [parallell for](http://docs.opencv.org/trunk/d7/dff/tutorial_how_to_use_OpenCV_parallel_for_.html), which is more universal than implicit OpenMP usage, but it hasn't critical sections unfortunately. Processing of every `poses[i]` seems to be independent from others. Why do you need `omp critical` here?